core/src/main/java/org/mskcc/cbio/oncokb/util/IndicatorUtils.java

```diff
@@ -170,60 +187,64 @@ public static IndicatorQueryResp processQuery(Query query, String geneStatus,
-                        Oncogenicity tmpOncogenicity = MainUtils.setToAlleleOncogenicity(Oncogenicity.getByEffect(oncogenicityEvidence.getKnownEffect()));
-                        if (tmpOncogenicity != null) {
-                            oncogenicity = tmpOncogenicity;
+                    // If there is no oncogenic info available for this variant, find oncogenicity from relevant variants
+                    if (oncogenicity == null || oncogenicity.equals(Oncogenicity.INCONCLUSIVE)) {
+                        oncogenicityEvidence = MainUtils.findHighestOncogenicEvidenceByEvidences(
+                            EvidenceUtils.getRelevantEvidences(query, source, geneStatus,
+                                Collections.singleton(EvidenceType.ONCOGENIC), null));
+                        if (oncogenicityEvidence != null) {
+                            Oncogenicity tmpOncogenicity = Oncogenicity.getByEffect(oncogenicityEvidence.getKnownEffect());
+                            if (tmpOncogenicity != null) {
+                                oncogenicity = tmpOncogenicity;
+                            }
                         }
                     }
-                }
 
-                // If there is no oncogenic info available for this variant, find oncogenicity from relevant variants
-                if (oncogenicity == null || oncogenicity.equals(Oncogenicity.INCONCLUSIVE)) {
-                    oncogenicityEvidence = MainUtils.findHighestOncogenicEvidenceByEvidences(
-                        EvidenceUtils.getRelevantEvidences(query, source, geneStatus,
-                            Collections.singleton(EvidenceType.ONCOGENIC), null));
                     if (oncogenicityEvidence != null) {
-                        Oncogenicity tmpOncogenicity = Oncogenicity.getByEffect(oncogenicityEvidence.getKnownEffect());
-                        if (tmpOncogenicity != null) {
-                            oncogenicity = tmpOncogenicity;
-                        }
+                        allQueryRelatedEvidences.add(oncogenicityEvidence);
                     }
-                }
 
-                if (oncogenicityEvidence != null) {
-                    allQueryRelatedEvidences.add(oncogenicityEvidence);
+                    // Only set oncogenicity if no previous data assigned.
+                    if (indicatorQuery.getOncogenic() == null && oncogenicity != null) {
+                        indicatorQuery.setOncogenic(oncogenicity.getOncogenic());
+                    }
                 }
 
-                // Only set oncogenicity if no previous data assigned.
-                if (indicatorQuery.getOncogenic() == null && oncogenicity != null) {
-                    indicatorQuery.setOncogenic(oncogenicity.getOncogenic());
+                if (hasTreatmentEvidence) {
+                    treatmentEvidences = EvidenceUtils.keepHighestLevelForSameTreatments(
+                        EvidenceUtils.getRelevantEvidences(query, source, geneStatus,
```

Comment:
Should we do the null check on the getRelevantEvidences() return before calling keepHighestLevelForSameTreatments()?

Reply:
getRelevantEvidences will always return an empty list if nothing has been found.